zookeeper/core/cli.py

```diff
@@ -7,6 +7,14 @@
 
 class ConfigParam(click.ParamType):
     def convert(self, str_value, param, ctx):
+        # Allow the syntax `--name_of_key` to set the boolean flag to `True`,
+        # and likewise `--no-name_of_key` to set the boolean flag to `False`.
+        if re.match("^--[\\w.]+$", str_value):
+            return str_value[2:], True
+        if re.match("^--no-[\\w.]+$", str_value):
+            return str_value[5:], False
```

Comment:
Isn't there something we could use from `click` to do this parsing?

Reply:
It's possible we could get it to work. This PR uses `ignore_unknown_options` as mentioned [here](https://click.palletsprojects.com/en/5.x/advanced/#forwarding-unknown-options) but I'm not sure how that interacts with the variadic argument which by default picks up all unmatched strings passed in.

Reply:
By which I mean, I know that this does work correctly but I'm not sure how a different solution where Click does the parsing would interact correctly.

Reply:
Makes sense :+1: